ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java

```diff
@@ -625,16 +694,18 @@ FindInfo findEntriesSince(FindToken token, long maxTotalSizeOfEntries) throws St
         // check journal
         List<JournalEntry> entries = journal.getEntriesSince(offsetToStart, storeToken.getInclusive());
         logger.trace("Journal based token, Time used to get entries: {}", (time.milliseconds() - startTimeInMs));
+        ConcurrentSkipListMap<Offset, IndexSegment> indexes = indexesRef.get();
```

Comment:
There aren't any issues. For your second comment, the snapshot is taken in the right places (just after looking at the journal).